File: japiwdbe/WdbeQVitList.java
/**
	* \file WdbeQVitList.h
	* Java API code for record of table TblWdbeQVitList
	* \copyright (C) 2018-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

package apiwdbe;

import org.w3c.dom.*;
import sbecore.*;

public class WdbeQVitList {
	
	public WdbeQVitList(
				int jnum
				, String sref
				, String Title
				, String stubVecRefWdbeMVector
			) {
		this.jnum = jnum;
		this.sref = sref;
		this.Title = Title;
		this.stubVecRefWdbeMVector = stubVecRefWdbeMVector;
	};

	public int jnum;
	public String sref;
	public String Title;
	public String stubVecRefWdbeMVector;
	
	public boolean readXML(
				Document doc
				, String basexpath
				, boolean addbasetag
			) {
		if (addbasetag) basexpath = Xmlio.checkUclcXPaths(doc, basexpath, "WdbeQVitList");

		if (Xmlio.checkXPath(doc, basexpath)) {
			sref = Xmlio.extractStringUclc(doc, basexpath, "sref", "srf", null, 0);
			Title = Xmlio.extractStringUclc(doc, basexpath, "Title", "tit", null, 0);
			stubVecRefWdbeMVector = Xmlio.extractStringUclc(doc, basexpath, "stubVecRefWdbeMVector", "vec", null, 0);
			
			return true;
		};

		return false;
	};
};

File: apiwdbe/WdbeQVitList.cpp
/**
	* \file WdbeQVitList.cpp
	* API code for table TblWdbeQVitList (implementation)
	* \copyright (C) 2016-2020 MPSI Technologies GmbH
	* \author Alexander Wirthmueller (auto-generation)
	* \date created: 5 Dec 2020
  */
// IP header --- ABOVE

#include "WdbeQVitList.h"

using namespace std;
using namespace Sbecore;
using namespace Xmlio;

/******************************************************************************
 class WdbeQVitList
 ******************************************************************************/

WdbeQVitList::WdbeQVitList(
			const uint jnum
			, const string sref
			, const string Title
			, const string stubVecRefWdbeMVector
		) {
	this->jnum = jnum;
	this->sref = sref;
	this->Title = Title;
	this->stubVecRefWdbeMVector = stubVecRefWdbeMVector;
};

bool WdbeQVitList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "WdbeQVitList");
	else
		basefound = checkXPath(docctx, basexpath);

	if (basefound) {
		extractStringUclc(docctx, basexpath, "sref", "srf", sref);
		extractStringUclc(docctx, basexpath, "Title", "tit", Title);
		extractStringUclc(docctx, basexpath, "stubVecRefWdbeMVector", "vec", stubVecRefWdbeMVector);
	};

	return basefound;
};

/******************************************************************************
 class ListWdbeQVitList
 ******************************************************************************/

ListWdbeQVitList::ListWdbeQVitList() {
};

ListWdbeQVitList::ListWdbeQVitList(
			const ListWdbeQVitList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQVitList(*(src.nodes[i]));
};

ListWdbeQVitList& ListWdbeQVitList::operator=(
			const ListWdbeQVitList& src
		) {
	nodes.resize(src.nodes.size(), NULL);
	for (unsigned int i = 0; i < nodes.size(); i++) nodes[i] = new WdbeQVitList(*(src.nodes[i]));

	return *this;
};

ListWdbeQVitList::~ListWdbeQVitList() {
	clear();
};

void ListWdbeQVitList::clear() {
	for (unsigned int i = 0; i < nodes.size(); i++) if (nodes[i]) delete nodes[i];
	nodes.resize(0);
};

bool ListWdbeQVitList::readXML(
			xmlXPathContext* docctx
			, string basexpath
			, bool addbasetag
		) {
	bool basefound;

	vector<unsigned int> ics;
	vector<bool> shorttags;

	WdbeQVitList* rec = NULL;

	string s;

	if (addbasetag)
		basefound = checkUclcXPaths(docctx, basexpath, basexpath, "ListWdbeQVitList");
	else
		basefound = checkXPath(docctx, basexpath);

	extractList(docctx, basexpath, "WdbeQVitList", "row", "jnum", ics, shorttags);

	clear();

	for (unsigned int i = 0; i < ics.size(); i++) {
		rec = new WdbeQVitList(ics[i]);

		if (shorttags[i]) s = basexpath + "/row[@jnum='" + to_string(ics[i]) + "']";
		else s = basexpath + "/WdbeQVitList[@jnum='" + to_string(ics[i]) + "']";

		if (rec->readXML(docctx, s)) nodes.push_back(rec);
		else delete rec;
	};

	return basefound;
};

